core/services/ocr2/plugins/promwrapper/plugin.go

```diff
@@ -0,0 +1,141 @@
+package promwrapper
+
+import (
+	"context"
+	"math/big"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/smartcontractkit/libocr/offchainreporting2/types"
+)
+
+var _ types.ReportingPlugin = &promPlugin{}
+
+var (
+	promQuery = promauto.NewHistogramVec(
```

Comment:
We can go with these metrics for now, I imagine we will add more (such as a transmission counter)

Reply:
Yeah we could definitely add more in the future. I think transmission counter specifically could be done using on chain logs and some custom metrics can be added in the off-chain coordinator as well.

Reply:
I wonder if a transmission counter can be added to the transmitter objects?

Reply:
I don't think all the prom metrics have to be in the plugin.

Reply:
@makramkd did you mean adding the transmitter counter on chain or something else?

Reply:
oh i see. yeah prom `counter` will be useful there to see rate of change for transmissions by transmitters